integration_tests/src/tests/prove_commit_niporep_test.rs

```diff
@@ -0,0 +1,339 @@
+        assert_eq!(activation_epoch, on_chain_sector.power_base_epoch);
+        assert!(on_chain_sector.flags.contains(SectorOnChainInfoFlags::SIMPLE_QA_POWER));
+    }
+
+    let deadline = deadline_state(v, &maddr, proving_deadline);
+    assert_eq!(deadline.live_sectors, sector_nos.len() as u64);
+}
+
+#[vm_test]
+pub fn prove_commit_ni_partial_success_not_required_test(v: &dyn VM) {
+    // Expectations depend on the correct unsealed CID for empty sector.
+    override_compute_unsealed_sector_cid(v);
+    let addrs = create_accounts(v, 3, &TokenAmount::from_whole(10_000));
+    let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P2_Feat_NiPoRep;
+    let (owner, worker, _, _) = (addrs[0], addrs[0], addrs[1], addrs[2]);
+    let worker_id = worker.id().unwrap();
+    let (maddr, _) = create_miner(
+        v,
+        &owner,
+        &worker,
+        seal_proof.registered_window_post_proof().unwrap(),
+        &TokenAmount::from_whole(8_000),
+    );
+    let miner_id = maddr.id().unwrap();
+    let policy = Policy::default();
+
+    // Onboard a batch of sectors
+    let seal_rand_epoch = v.epoch();
+    let activation_epoch = seal_rand_epoch + policy.max_prove_commit_ni_randomness_lookback / 2;
+    let expiration = activation_epoch + policy.min_sector_expiration + 1;
+    let first_sector_number: SectorNumber = 100;
+    let sector_nos = [
+        first_sector_number,
+        first_sector_number + 1,
+        first_sector_number + 2,
+        first_sector_number + 3,
+        first_sector_number + 4,
+    ];
+    let proving_deadline = 7;
+
+    let mut sectors_info: Vec<SectorNIActivationInfo> = sector_nos
+        .iter()
+        .map(|sector_number| SectorNIActivationInfo {
+            sealing_number: *sector_number,
+            sealer_id: miner_id,
+            sector_number: *sector_number,
+            sealed_cid: make_sealed_cid(format!("sn: {}", sector_number).as_bytes()),
+            seal_rand_epoch,
+            expiration,
+        })
+        .collect();
+
+    sectors_info[0].seal_rand_epoch = activation_epoch;
+
+    let invalid_sector_nos = sector_nos.iter().take(1).copied().collect::<Vec<_>>();
+    let valid_sector_nos = sector_nos.iter().skip(1).copied().collect::<Vec<_>>();
+    let valid_sectors_info = sectors_info.iter().skip(1).cloned().collect::<Vec<_>>();
+
+    // Prove-commit NI-PoRep
+    let aggregate_proof = RawBytes::new(vec![1, 2, 3, 4]);
+    let params = ProveCommitSectorsNIParams {
+        sectors: sectors_info.clone(),
+        seal_proof_type: RegisteredSealProof::StackedDRG32GiBV1P2_Feat_NiPoRep,
+        aggregate_proof,
+        aggregate_proof_type: RegisteredAggregateProof::SnarkPackV2,
+        proving_deadline,
+        require_activation_success: false,
+    };
+
+    v.set_epoch(activation_epoch);
+
+    apply_ok(
+        v,
+        &worker,
+        &maddr,
+        &TokenAmount::zero(),
+        MinerMethod::ProveCommitSectorsNI as u64,
+        Some(params.clone()),
+    );
```

Comment:
```suggestion
    let ret_br: BatchReturn = apply_ok(
        v,
        &worker,
        &maddr,
        &TokenAmount::zero(),
        MinerMethod::ProveCommitSectorsNI as u64,
        Some(params.clone()),
    )
    .deserialize()
    .unwrap();

    assert_eq!(ret_br.len(), 2);
    assert!(ret_bf.ok());
    assert_eq!(ret_bf.size(), 4);
    // something more to inspect codes and/or successes
```

would be really nice to check the return value in this, and the successful case in `prove_commit_ni_whole_success_test` (I'm ~guessing at the suggested code here based on the very few [unfortunately] instances of checking, such as in integration_tests/src/tests/replica_update_test.rs).